common:
  # actuated_joint_names: [r_shoulder_pan_joint, r_shoulder_lift_joint, r_upper_arm_roll_joint, r_elbow_flex_joint, r_forearm_roll_joint, r_wrist_flex_joint, r_wrist_roll_joint]
  actuated_joint_names: [
    r_shoulder_pan_joint,
    r_shoulder_lift_joint,
    r_upper_arm_roll_joint,
    r_elbow_flex_joint,
    r_forearm_roll_joint,
    r_wrist_flex_joint,
    r_wrist_roll_joint,
    l_shoulder_pan_joint,
    l_shoulder_lift_joint,
    l_upper_arm_roll_joint,
    l_elbow_flex_joint,
    l_forearm_roll_joint,
    l_wrist_flex_joint,
    l_wrist_roll_joint
    ]
  max_joint_velocity: 1.0

  # Controllers
  position_loop_gains:
    r_upper_arm_roll_joint:
      p: 6
      i: 600
      d: 6
      i_clamp: 4

    r_shoulder_pan_joint:
      p: 2400
      i: 800
      d: 18
      i_clamp: 4

    r_shoulder_lift_joint:
      p: 1200
      i: 700
      d: 10
      i_clamp: 4

    r_forearm_roll_joint:
      p: 300
      i: 300
      d: 6
      i_clamp: 2

    r_elbow_flex_joint:
      p: 700
      i: 450
      d: 4
      i_clamp: 4

    r_wrist_flex_joint:
      p: 300
      i: 300
      d: 4
      i_clamp: 2

    r_wrist_roll_joint:
      p: 300
      i: 300
      d: 4
      i_clamp: 2

    l_upper_arm_roll_joint:
      p: 6
      i: 600
      d: 6
      i_clamp: 4

    l_shoulder_pan_joint:
      p: 2400
      i: 800
      d: 18
      i_clamp: 4

    l_shoulder_lift_joint:
      p: 1200
      i: 700
      d: 10
      i_clamp: 4

    l_forearm_roll_joint:
      p: 300
      i: 300
      d: 6
      i_clamp: 2

    l_elbow_flex_joint:
      p: 700
      i: 450
      d: 4
      i_clamp: 4

    l_wrist_flex_joint:
      p: 300
      i: 300
      d: 4
      i_clamp: 2

    l_wrist_roll_joint:
      p: 300
      i: 300
      d: 4
      i_clamp: 2
  velocity_loop_gains:
    r_upper_arm_roll_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 2
      feedforward_gain: 30

    r_shoulder_pan_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 1
      feedforward_gain: 20

    r_shoulder_lift_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 7.5
      feedforward_gain: 20

    r_forearm_roll_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 20

    r_elbow_flex_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 20

    r_wrist_flex_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 1

    r_wrist_roll_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 1

    l_upper_arm_roll_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 2
      feedforward_gain: 30

    l_shoulder_pan_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 1
      feedforward_gain: 20

    l_shoulder_lift_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 7.5
      feedforward_gain: 20

    l_forearm_roll_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 20

    l_elbow_flex_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 20

    l_wrist_flex_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 1

    l_wrist_roll_joint:
      p: 1
      i: 0
      d: 0
      i_clamp: 10
      feedforward_gain: 1

manipulation_joint_controller:
  type: pr2_joint_position_controllers/ManipulationJointController
  feedback_rate: 1
  receive_position_reference: False
  control_timeout: 1

approach_joint_controller:
  type: pr2_joint_position_controllers/ApproachJointController
  feedback_rate: 1
  receive_position_reference: False
  control_timeout: 1

move_joint_controller:
  type: pr2_joint_position_controllers/MoveJointController
  feedback_rate: 100
  receive_position_reference: True
  control_timeout: 1

folding_joint_controller:
  type: pr2_joint_position_controllers/FoldingJointController
  feedback_rate: 1
  receive_position_reference: True
  control_timeout: 1
  
mechanism_joint_controller:
  type: pr2_joint_position_controllers/MechanismJointController
  feedback_rate: 50
  receive_position_reference: True
  control_timeout: 1
